crates/vault-registry/src/types.rs

```diff
@@ -441,40 +441,59 @@ impl<T: Config> RichVault<T> {
         // Note: slashing of collateral must be called where this function is called (e.g. in Redeem)
     }
 
+    pub(crate) fn slash_collateral_and_transfer(&mut self, amount: Collateral<T>) -> DispatchResult {
+        self.slash_collateral(amount)?;
+        let vault_id = self.id();
```

Comment:
It looks like from here it does the same as `transfer_funds` - can we use that instead? I would prefer to keep the transferring logic in a single place rather than spread out over the entire file. That way we are sure that variables like the `backing_collateral` are always correctly updated

Reply:
> The problem with using methods defined on the `Pallet` from the `RichVault` is that it won't also mutate the internal state - i.e. we would need to indeterminately refresh the `data` inner from storage.

But the code starting from line 446 doesn't mutate internal state. So in this function, keep `slash_collateral`, and then call `transfer_funds`. 

I get your point about internal state being a problem. But we still should try to do what we can to keep the collateral handling as centralized as possible

Reply:
Point taken, I will rewrite to call `transfer_funds` as before.

Reply:
Actually I realize we can't do this because `transfer_funds` also calls `slash_backing_collateral` so I will have to leave this for now. I would probably favor refactoring `transfer_funds` in the future anyway to be more specific.

Reply:
Haha I would actually prefer `transfer_funds` to be as general as possible - ideally it'd be the only function that touches collateral